### DISCUSSION TITLE
Highlight occurrences of selected text

### DISCUSSION THREAD

**Discussion by eauw** [zed-industries/zed#9238] [Feature Requests]
Hello,
I wonder if I can't find the configuration for that or if this feature just isn't there (yet).
I'm talking about highlighting other occurrences of selected text. In some editors all other matching strings are being highlighted even on just clicking on it.

**Comment by nstlgy** [comment]
This is an already available feature in zed. You simply double-click on the word you want to find other occurrences of, and it should highlight the rest automatically.

**Reply by eauw** [reply]
Thank you. I assumed that and now I just tried it in a different project and it works in a Flutter project. But it doesn't work in my Svelte/Typescript projects. Any idea?

**Reply by Moshyfawn** [reply]
Which syntax nodes do you expect it to highlight?

**Reply by eauw** [reply]
@Moshyfawn I'm talking about variables. I added an image of some example. As you can see I marked the variable at the top and the other occurrences are not highlighted.
<img width="1178" alt="Screenshot 2024-03-14 at 10 14 38" src="https://github.com/zed-industries/zed/assets/2536825/47fc0407-c0e5-41b5-a3d0-03133f6fdffd">

**Reply by shinebayar-g** [reply]
It doesn't.

**Reply by PHLAK** [reply]
This is definitely working.

<img src="https://github.com/user-attachments/assets/bf98dc56-703f-4de8-9fc6-b66e5edd8b2b" width="50%">

Maybe this is an issue with your theme?

**Reply by shinebayar-g** [reply]
![image](https://github.com/user-attachments/assets/f3df46ef-88f7-4d14-ada2-a6da2c2cbd7c)

Default theme?

**Reply by lanternlogic** [reply]
it seems to work with certain files and even in those files certain symbols. 
In VSCode (and IIRC Sublime), selecting _any_ word/symbol in any file will select other occurrences of that thing.

**Comment by eauw** [comment]
Anyone?

**Reply by baldwindavid** [reply]
I think you're referring to something like the "editor.occurrencesHighlight" setting in vscode. I'm unaware of any setting like that in Zed. I see others reference a double-click for occurrences. I don't click things anyway, but fwiw don't see that working in an elixir file. The closest I can think of are all keyboard-driven. In vim, I'd do `*` to select the next match which will highlight all, `/` to search for specific text, or even `g a` which will select all occurrences of the current word in the buffer via the `editor: select all matches` command.

**Reply by Cervajz** [reply]
I confirm that this does not work for Elixir files.

In javascript I don't have to double-click or select the variable, it works automatically.

**Comment by Kociamber** [comment]
I can also confirm it for Elixir, double clicking on a string does not highlight all the occurrences in the code. It's a small feature but I find it extremely useful (especially when highlight occurs in all open tabs).

**Comment by barkerja** [comment]
Another Elixir user here. Was confused by this behavior as I make the switch over from VSCode to Zed.

**Comment by lanternlogic** [comment]
I'm also confused by Zed's behaviour on this. Anyone notice / is bothered by Zed's inability to highlight other occurrences of words in text files, or anything they don't fully "support" yet?
I find that super annoying, unless I've something is misconfigured in my setup?

**Comment by jaydenseric** [comment]
See https://github.com/zed-industries/zed/issues/12635 .

**Comment by AndrStnz** [comment]
In BASH scripts it is also not highlighting "same selected text", this is such an important comfort feature...

**Comment by defkode** [comment]
CMD+F works for me

**Comment by ItaloCunhaBueno** [comment]
Does not work with python files too.

**Comment by GrabbenD** [comment]
Don't forget to drop a :+1: in https://github.com/zed-industries/zed/issues/12635 (`Highlight all matching occurrences of text in selection`) for this feature to gain visibility in [Top-Ranking Issues 📊](#6953) tracker

_(Sorry for double post but this is the first result on Google)_

**Comment by smitbarmase** [comment]
This feature recently landed in https://github.com/zed-industries/zed/pull/24835. Thank you for your patience. Closing this.

**Reply by barkerja** [reply]
A question I've asked on the Discord but never got an answer for .. is this new addition configurable as part of the theme? (E.g., can I control the color of the highlights?)

🙇🏻

**Reply by smitbarmase** [reply]
Yes, you can override the theme using [theme overrides](https://zed.dev/docs/themes#theme-overrides). Specifically, try experimenting with these values:

```jsonc
    /// Read-access of a symbol, like reading a variable.
    ///
    /// A document highlight is a range inside a text document which deserves
    /// special attention. Usually a document highlight is visualized by changing
    /// the background color of its range.
    #[serde(rename = "editor.document_highlight.read_background")]
    pub editor_document_highlight_read_background: Option<String>,

    /// Read-access of a symbol, like reading a variable.
    ///
    /// A document highlight is a range inside a text document which deserves
    /// special attention. Usually a document highlight is visualized by changing
    /// the background color of its range.
    #[serde(rename = "editor.document_highlight.write_background")]
    pub editor_document_highlight_write_background: Option<String>,
```

I wonder, though, how granular it is, like the background for a specific type of highlight? Let me know.

**Reply by barkerja** [reply]
Thank you!

**Reply by AndrStnz** [reply]
Here is my working Theme Override. Add it to your `settings.json`
```
  "experimental.theme_overrides": {
    "editor.background": "#222",
    "editor.document_highlight.read_background": "#992",
    "editor.document_highlight.write_background": "#900",
    "syntax": {
      "comment": {
        "font_style": "italic"
      },
      "comment.doc": {
        "font_style": "italic"
      }
    }
  }
```

**Comment by julien** [comment]
Is there any way to disable this?  
I'm using `"selection_highlight": false`, but that doesn't seem to have any effect in .go, .c or .rs files. Thanks

**Reply by eugenesvk** [reply]
think you also need to override your theme, set color tranparency to `00` as that's a separate non-selection related type of highlight
```json
"editor.document_highlight.read_background"    	: "#00ff8500",
"editor.document_highlight.write_background"   	: "#ff847400",
```

**Reply by julien** [reply]
Perfect, thanks!

**Comment by victorcl** [comment]
This works as expected for me when I double click a word, but would it be possible for it to automatically highlight the occurrences also when the cursor is on the word without me having to double click it?
